src/Symfony/Component/Notifier/Bridge/Ntfy/NtfyTransportFactory.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Ntfy;
+
+use Symfony\Component\Notifier\Exception\UnsupportedSchemeException;
+use Symfony\Component\Notifier\Transport\AbstractTransportFactory;
+use Symfony\Component\Notifier\Transport\Dsn;
+use Symfony\Component\Notifier\Transport\TransportInterface;
+
+/**
+ * @author Mickael Perraud <mikaelkael.fr@gmail.com>
+ */
+final class NtfyTransportFactory extends AbstractTransportFactory
+{
+    public function create(Dsn $dsn): TransportInterface
+    {
+        if ('ntfy' !== $dsn->getScheme()) {
+            throw new UnsupportedSchemeException($dsn, 'ntfy', $this->getSupportedSchemes());
+        }
+
+        $host = 'default' === $dsn->getHost() ? null : $dsn->getHost();
+        $topic = substr($dsn->getPath(), 1);
+        $transport = (new NtfyTransport($topic))->setHost($host);
+        if (!empty($port = $dsn->getPort())) {
+            $transport->setPort($port);
+        }
+
+        $secureHttp = $dsn->getOption('secureHttp', true);
+        if (\in_array($secureHttp, [0, false, 'false', 'off', 'no'])) {
+            $transport->setSecureHttp(false);
```

Comment:
Please set them via the constructor like done in the other bridges and remove the setters/getters from the transport, thanks

Reply:
Took me a while to figure it out but I understand :wink: